### PR TITLE
Add support for Tuya TS0046

### PR DESCRIPTION
**Pull request by SkyaTura**
Made in colab with @MattWestb and @javicalle _(mostly by them)_ as discussed in #2034

fixes: #2034

### CODE REVIEW

**Review by MattWestb — APPROVED** [review]
Is adding new devices that was not paired before (or retested and battery taken out and new paring = not in ZHA as device) working OK ?
I think so then you have getting the tuya magic nice implanted.

Is it working OK with moving the OnOff cluster to out on all endpoints and the not working switches is away ?

For my its looks very good only that i dont have one for testing the quirk.

**Comment by MattWestb** [comment]
Pleas add `fixes: #2034` in  the opening post so its closing the initial issue for the device request :-))

File: zhaquirks/tuya/ts0046.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        (SHORT_PRESS, BUTTON_5): {ENDPOINT_ID: 5, COMMAND: SHORT_PRESS},
+        (LONG_PRESS, BUTTON_5): {ENDPOINT_ID: 5, COMMAND: LONG_PRESS},
+        (DOUBLE_PRESS, BUTTON_5): {ENDPOINT_ID: 5, COMMAND: DOUBLE_PRESS},
+        (SHORT_PRESS, BUTTON_6): {ENDPOINT_ID: 6, COMMAND: SHORT_PRESS},
+        (LONG_PRESS, BUTTON_6): {ENDPOINT_ID: 6, COMMAND: LONG_PRESS},
+        (DOUBLE_PRESS, BUTTON_6): {ENDPOINT_ID: 6, COMMAND: DOUBLE_PRESS},
+    }
+
+
+class TuyaSmartRemote0046TOPlusA(EnchantedDevice, Tuya6ButtonTriggers):
```
Comment (javicalle) [inline zhaquirks/tuya/ts0046.py:54]: Maybe just `TuyaSmartRemote0046TO` or `TuyaSmartRemote0046`?

**Review by javicalle — APPROVED** [review]
LGTM

**Pull request merged**